xilem/src/view/sized_box.rs

```diff
@@ -209,3 +221,84 @@ struct BorderStyle {
     width: f64,
     color: Color,
 }
+
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct Padding {
+    top: f64,
+    trailing: f64,
+    bottom: f64,
+    leading: f64,
+}
+
+impl Padding {
+    pub const fn new(top: f64, trailing: f64, bottom: f64, leading: f64) -> Self {
+        Self {
+            top,
+            trailing,
+            bottom,
+            leading,
+        }
+    }
+
+    pub const ZERO: Padding = Padding::all(0.);
+
+    pub const fn all(padding: f64) -> Self {
+        Self::new(padding, padding, padding, padding)
+    }
+
+    pub const fn horizontal(padding: f64) -> Self {
+        Self::new(0., padding, 0., padding)
+    }
+
+    pub const fn vertical(padding: f64) -> Self {
+        Self::new(padding, 0., padding, 0.)
+    }
+
+    pub const fn top(padding: f64) -> Self {
+        Self::new(padding, 0., 0., 0.)
+    }
+
+    pub const fn trailing(padding: f64) -> Self {
+        Self::new(0., padding, 0., 0.)
+    }
+
+    pub const fn bottom(padding: f64) -> Self {
+        Self::new(0., 0., padding, 0.)
+    }
+
+    pub const fn leading(padding: f64) -> Self {
+        Self::new(0., 0., 0., padding)
+    }
+}
+
+impl From<f64> for Padding {
+    fn from(value: f64) -> Self {
+        Self::all(value)
+    }
+}
+
+// Follows CSS padding order for 4 values (top, trailing, bottom, leading)
+impl From<(f64, f64, f64, f64)> for Padding {
```

Comment:
We need to document why we use a different type to `Insets`, somewhere. Probably directly on `Padding`.

(That reason is so that these impls can exist)

Reply:
I moved the Padding struct to Masonry and completely removed the dependency on Insets.